Tài liệu + source code về dockerize

nguyenmphu opened the pull request:
Demo app sử dụng framework flask

Commit:
Tài liệu + source code về dockerize

Demo app sử dụng framework flask

cuongtransc commented on docker/Dockerize Flask App/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+# Format: FROM    repository[:version]
+FROM       ubuntu:latest
```
- Không nên sử dụng tag latest.
- Nên sử dụng tag version, ví dụ: ubuntu:16.04

cuongtransc commented on docker/Dockerize Flask App/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+# Format: FROM    repository[:version]
+FROM       ubuntu:latest
+RUN apt-get update -y
+RUN apt-get install -y python-pip python-dev build-essential
```
Dùng python2 ok. Nhưng nên dùng python3.

cuongtransc commented on docker/Dockerize Flask App/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+RUN apt-get install -y python-pip python-dev build-essential
+COPY . /app
+WORKDIR /app
+RUN pip install -r requirements.txt
```
Copy toàn bộ source code, sau đó mới install requirements.
-> Mỗi khi update source code lại cài lại requirements, mặc dù có thể requirements không thay đổi.

Nên đổi lại thứ tự kiểu như:

```
COPY requirements /tmp/requirements
RUN pip3 install -r /tmp/requirements/base.txt
RUN pip3 install -r /tmp/requirements/production.txt

COPY . /usr/src/app
WORKDIR /usr/src/app
```

cuongtransc reviewed (commented):
Cần update Dockerfile

Commit:
Update

- Sử dụng image gốc ubuntu:16.04
- Sử dụng python 3 thay cho python 2
- Thay đổi thứ tự trong build image: cài đặt requirement trước khi copy source code

nguyenmphu commented:
Em đã update lại Dockerfile
